libs/core/functional/include/hpx/functional/tag_dispatch.hpp

```diff
@@ -184,13 +182,25 @@ namespace hpx { namespace functional {
         };
     }    // namespace detail
 
+    // CUDA versions less than 11.2 have a template instantiation bug which
+    // leaves out certain template arguments and leads to us not being able to
+    // correctly check this condition. We default to the more relaxed
+    // noexcept(true) to not falsely exclude correct overloads. However, this
+    // may lead to noexcept(false) overloads falsely being candidates.
+#if !defined(HPX_CUDA_VERSION) || (HPX_CUDA_VERSION >= 1102)
```

Comment:
Can this be checked by a feature test instead of relying on a concrete version?

Reply:
My personal preference is the explicit CUDA version check, because this is not about a feature, this is a pure bug in nvcc. In addition, the workaround is very specific to `tag_dispatch` and does not need to be applied everywhere we check for noexcept.

Reply:
Ok, fair enough.